### PR TITLE
Update terraform to 0.13.5 version

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/12286

### CODE REVIEW

**Comment by corest** [comment]
No dramatic changes.
Most of the updates are related to warning abut interpolated strings
Few changes regarding how 3rd party providers are managed now

**Review by whites11 — APPROVED** [review]

**Review by kopiczko — APPROVED** [review]

**Review by calvix — APPROVED** [review]

File: modules/aws/vault/versions.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+terraform {
+  required_providers {
+    aws = {
+      source = "hashicorp/aws"
```
Comment (mazzy89) [inline modules/aws/vault/versions.tf:4]: small thing: below we pin providers, here no. is this wanted? should we pin providers everywhere?

**Review by mazzy89 — CHANGES_REQUESTED** [review]
tiny question about pinning providers

**Comment by mazzy89** [comment]
arrived too late 😇

**Pull request merged**